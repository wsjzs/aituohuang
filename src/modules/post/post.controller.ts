import { Controller, Get, Query } from '@nestjs/common'

import { HTTPDecorators } from '~/common/decorators/http.decorator'
import { BizException } from '~/common/exceptions/business.excpetion'
import { ErrorCodeEnum } from '~/constants/error-code.constant'
import { PagerDto } from '~/shared/dto/pager.dto'

import { PostService } from './post.service'

@Controller('posts')
export class PostController {
  constructor(private readonly service: PostService) {}

  @HTTPDecorators.Paginator
  @Get('/')
  async gets(@Query() query: PagerDto) {
    const { page, select, size } = query
    return this.service.model.paginate(
      {},
      {
        page,
        limit: size,
        select,
        lean: true,
      },
    )
  }

  @Get('/*')
  async notFound() {
    throw new BizException(ErrorCodeEnum.PostNotFoundError)
  }
}
