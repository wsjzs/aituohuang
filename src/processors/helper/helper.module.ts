import { Global, Module, Provider } from '@nestjs/common'

import { HttpService } from './helper.http.service'

const providers: Provider<any>[] = [HttpService]

@Module({
  imports: [],
  providers,
  exports: providers,
})
@Global()
export class HelperModule {}
