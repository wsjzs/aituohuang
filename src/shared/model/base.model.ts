// import LeanId from 'mongoose-lean-id'
// import { default as mongooseLeanVirtuals } from 'mongoose-lean-virtuals'
import Paginate from 'mongoose-paginate-v2'

import { modelOptions, plugin } from '@typegoose/typegoose'

// @plugin(mongooseLeanVirtuals)
// @plugin(LeanId)
@plugin(Paginate)
@modelOptions({
  schemaOptions: {
    toJSON: { virtuals: true },
    toObject: { virtuals: true },
    timestamps: {
      createdAt: 'created',
      updatedAt: false,
    },
  },
})
export class BaseModel {
  created?: Date

  id?: string

  static get protectedKeys() {
    return ['created', 'id', '_id']
  }
}
