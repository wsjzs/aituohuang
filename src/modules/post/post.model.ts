import { IsNotEmpty, IsString } from 'class-validator'

import { Severity, index, modelOptions, prop } from '@typegoose/typegoose'

import { BaseModel } from '~/shared/model/base.model'

@index({ slug: 1 })
@index({ modified: -1 })
@index({ text: 'text' })
@modelOptions({ options: { customName: 'Post', allowMixed: Severity.ALLOW } })
export class PostModel extends BaseModel {
  @prop({ trim: true, unique: true, required: true })
  @IsString()
  @IsNotEmpty()
  slug: string

  @prop({ trim: true, required: true })
  @IsString()
  text: string

  @prop({ trim: true, required: true })
  @IsString()
  title: string
}
