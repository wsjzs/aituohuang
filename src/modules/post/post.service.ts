import { Injectable } from '@nestjs/common'

import { InjectModel } from '~/transformers/model.transformer'

import { PostModel } from './post.model'

@Injectable()
export class PostService {
  constructor(
    @InjectModel(PostModel) public readonly model: MongooseModel<PostModel>,
  ) {}
}
