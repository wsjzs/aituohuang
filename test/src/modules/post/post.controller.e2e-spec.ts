import { createE2EApp } from 'test/helper/create-e2e-app'

import { PostController } from '~/modules/post/post.controller'
import { PostModel } from '~/modules/post/post.model'
import { PostService } from '~/modules/post/post.service'

describe('Test PostController E2E', () => {
  const proxy = createE2EApp({
    controllers: [PostController],
    providers: [PostService],
    models: [PostModel],
    async pourData(modelMap) {
      const { model } = modelMap.get(PostModel)
      await model.create([
        {
          title: 'test1',
          text: 'test1',
          slug: 'test1',
        },
        {
          title: 'test2',
          text: 'test2',
          slug: 'test2',
        },
      ])

      return async () => {
        await model.deleteMany({})
      }
    },
  })

  test('GET /posts', async () => {
    const app = proxy.app

    const res = await app.inject({
      method: 'GET',
      url: '/posts',
    })

    expect(res.statusCode).toBe(200)
    const result = res.json()
    expect(result.data.length).toBe(2)
  })
})
