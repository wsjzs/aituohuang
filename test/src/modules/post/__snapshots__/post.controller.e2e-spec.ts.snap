// Vitest Snapshot v1

exports[`Test PostController E2E > GET /posts 1`] = `
{
  "data": [
    {
      "created": "2023-01-28T08:25:17.105Z",
      "id": "63d4dbed3617841ef23cf4a3",
      "slug": "test1",
      "text": "test1",
      "title": "test1",
    },
    {
      "created": "2023-01-28T08:25:17.105Z",
      "id": "63d4dbed3617841ef23cf4a4",
      "slug": "test2",
      "text": "test2",
      "title": "test2",
    },
  ],
  "pagination": {
    "current_page": 1,
    "has_next_page": false,
    "has_prev_page": false,
    "size": 10,
    "total": 2,
    "total_page": 1,
  },
}
`;
